Add Awesome Brain Manager plugin

JuckZ opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/JuckZ/awesome-brain-manager

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
This plugin does to many things at once, and some of them are just straight up taken from other plugins, would recommend splitting it up into seperate ones.

Please credit the plugins you have taken code from properly,
I can see that some of it is from the reminder plugin for example.

[Copyright (c) [year] [fullname]](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/License#L3)
Replace these placeholders.

[- 🎨 More and more elaborate designs will be added, and your obsidian will be different](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/README_en.md#L48-L49)
looks like a bad translation

[import moment from 'moment';](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L23)
Already included in Obsidian, can import with `import { moment } from "obsidian";`


[/* An asynchronous sleep function. To use it you must `await` as it hands](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L702-L709)
The Obsidian API already brings a function with the same name, that does the same thing.

[this.app.workspace.getUnpinnedLeaf();](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L741)
Deprecated method, use `getLeaf(false)` instead.

[private async addACheck(path, filename, time, content) {](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L745-L764)
Would recommend the new `app.adapter.process` API, which is atomic,
<https://github.com/obsidianmd/obsidian-api/blob/bceb489fc25ceba5973119d6e57759d64850f90d/obsidian.d.ts#L749-L752>

[await app.vault.adapter.write(normalizedPath, newFileContent.content);](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L754)
It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future. Instead, you should use the reference to App that you have from your plugin instance.


[id: 'awesome-brain-manager-check-in',](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L828)
Obsidian will automatically prefix the command id with the plugin id,
you don't need to include it yourself.

[import { App, TFile, TFolder } from 'obsidian';](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/utils/file.ts#L1-L2)
Please use only one import style.

[notify](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/utils/request.ts#L39)
What is this for?

JuckZ commented:
Thank you for taking the time to perform a code review and provide such detailed feedback. I have addressed each of your suggestions and would appreciate it if you could review my code again.

1. This plugin does to many things at once, and some of them are just straight up taken from other plugins, would recommend splitting it up into seperate ones.
I have deleted many redundant codes and functions, and now it is mainly pomodoro and simple special effects.

2. Please credit the plugins you have taken code from properly,
Yes, of course, I should put my thanks in a prominent position. [REDEME#Thanks](https://github.com/JuckZ/awesome-brain-manager/blob/master/README_en.md#-thanks)

3. [Copyright (c) [year] [fullname]](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/License#L3)
I have replaced these placeholders. [License](https://github.com/JuckZ/awesome-brain-manager/blob/master/License)

4.[- 🎨 More and more elaborate designs will be added, and your obsidian will be different](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/README_en.md#L48-L49)
I have tried my best to retranslate it.

5. [import moment from 'moment';](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L23)
I have imported 'moment' from obsidian instead.

6. [/* An asynchronous sleep function. To use it you must await as it hands](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L702-L709)
Done.

7. [this.app.workspace.getUnpinnedLeaf();](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L741)
Done.

8. [private async addACheck(path, filename, time, content) {](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L745-L764)
Would recommend the new app.adapter.process API, which is atomic,
https://github.com/obsidianmd/obsidian-api/blob/bceb489fc25ceba5973119d6e57759d64850f90d/obsidian.d.ts#L749-L752
Done.

9. [await app.vault.adapter.write(normalizedPath, newFileContent.content);](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L754)
Done.

10. [id: 'awesome-brain-manager-check-in',](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/main.ts#L828)
Done.

11. [import { App, TFile, TFolder } from 'obsidian';](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/utils/file.ts#L1-L2)
Done.

12. [notify](https://github.com/JuckZ/awesome-brain-manager/blob/fbf9688b5b9ad07e38f1aa5efba40393ba29ec11/src/utils/request.ts#L39)
Due to limitations in the Electron API, it is not possible to receive message notifications on mobile devices. However, this functionality can be achieved using the ntfy tool, which provides an official service address that users can utilize. The official website for ntfy is https://ntfy.sh/.
@joethei
